packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -284,6 +285,45 @@ abstract class ImageProvider<T> {
     return stream;
   }
 
+  /// Evicts an entry from the image cache.
+  ///
+  /// Returns a [Future] which indicates whether the value was successfully
+  /// removed.
+  ///
+  /// The [ImageProvider] used does not need to be the same instance that was
+  /// passed to an [Image] widget, but it does need to create a key which is
+  /// equal to one.
+  ///
+  /// The [cache] is optional and defaults to the global image cache.
+  ///
+  /// The [configuration] is optional and defaults to
+  /// [ImageConfiguration.empty].
+  ///
+  /// ## Sample code
+  ///
+  /// The following sample code shows how an image loaded using the [Image]
+  /// widget can be evicted using a [NetworkImage] with a matching url.
+  ///
+  /// ```dart
+  /// class MyWidget extends StatelessWidget {
+  ///   final String url = '...';
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new Image.network(url);
+  ///   }
+  ///
+  ///   void evictImage() {
+  ///     final NetworkImage provider = new NetworkImage(url);
+  ///     provider.evict().then<void>((bool value) => debugPrint('removed image!'));
+  ///   }
+  /// }
+  /// ```
+  Future<bool> evict({ImageCache cache, ImageConfiguration configuration = ImageConfiguration.empty}) {
+    cache ??= imageCache;
+    return obtainKey(configuration).then<bool>(imageCache.evict);
+  }
```

Comment:
fwiw, this method (and the sample code) could be written more idiomatically using `async` and `await`:

```dart
void evictImage() async {
  final NetworkImage provider = new NetworkImage(url);
  if (await provider.evict())
    debugPrint('removed image!');
}

...

 Future<bool> evict({ImageCache cache, ImageConfiguration configuration = ImageConfiguration.empty}) async {
    cache ??= imageCache;
    return cache.evict(await obtainKey(configuration));
  }
￼```

(also, evictImage should probably only print if `value` is true, and you should add a test that checks that `evict` honors the `cache` argument)